Public/Src/Engine/Scheduler/PipExecutor.cs

```diff
@@ -1879,28 +1879,54 @@ private static void ReportFileAccesses(ExecutionResult processExecutionResult, F
         ///   a non-null result is returned.
         /// - If cache lookup fails (i.e., the result is inconclusive due to failed hashing, etc.), a null result is returned.
         /// </summary>
-        public static Task<RunnableFromCacheResult> TryCheckProcessRunnableFromCacheAsync(
+        public static async Task<RunnableFromCacheResult> TryCheckProcessRunnableFromCacheAsync(
             ProcessRunnablePip processRunnable,
             PipExecutionState.PipScopeState state,
             CacheableProcess cacheableProcess)
         {
-            return TryCheckProcessRunnableFromCacheAsync(
+            BoxRef<PipCacheMissEventData> pipCacheMiss = new PipCacheMissEventData
+            {
+                PipId = processRunnable.PipId,
+                CacheMissType = PipCacheMissType.Invalid,
+            };
+
+            var runnableFromCacheResult = await TryCheckProcessRunnableFromCacheAsync(
                 processRunnable,
                 state,
                 cacheableProcess,
                 computeWeakFingerprint: () => new WeakContentFingerprint(cacheableProcess.ComputeWeakFingerprint().Hash),
-                canAugmentWeakFingerprint: processRunnable.Process.AugmentWeakFingerprintPathSetThreshold(processRunnable.Environment.Configuration.Cache) > 0);
+                pipCacheMiss,
+                canAugmentWeakFingerprint: processRunnable.Process.AugmentWeakFingerprintPathSetThreshold(processRunnable.Environment.Configuration.Cache) > 0,
+                isWeakFingerprintAugmented: false);
+
+            if (!runnableFromCacheResult.CanRunFromCache)
+            {
+                Contract.Assert(pipCacheMiss.Value.CacheMissType != PipCacheMissType.Invalid, "Must have valid cache miss reason");
+                processRunnable.Environment.Counters.IncrementCounter((PipExecutorCounter)pipCacheMiss.Value.CacheMissType);
+
+                Logger.Log.ScheduleProcessPipCacheMiss(

```

Comment:
>ScheduleProcessPipCacheMiss [](start = 27, length = 27)

We already have counters for how many pips are misses due to the various reasons. Should we just add the CacheMissType to this log message and have the one log event? The discrete events below don't really add any more information and slimming down redundant information from our log file is goodness.

Reply:
Discussed offline

---
In reply to: [347055628](https://github.com/microsoft/BuildXL/pull/1205#discussion_r347055628) [](ancestors = 347055628)